components/sysfsGpio/gpioSysfsUtils.c

```diff
@@ -28,6 +28,37 @@
 #define MAX_PIN_NUMBER 64
 #define MIN_PIN_NUMBER 1
 
+//--------------------------------------------------------------------------------------------------
+/**
+ * Remove the change callback for the given GPIO
+ */
+//--------------------------------------------------------------------------------------------------
+static void RemoveChangeCallback
+(
+    gpioSysfs_GpioRef_t gpioRef  ///< GPIO to remove the change callback from
+)
+{
+    // If there is an fd monitor then stop it
+    if (gpioRef->fdMonitor != NULL)
+    {
+        LE_DEBUG("Stopping fd monitor");
+        const int fd = le_fdMonitor_GetFd(gpioRef->fdMonitor);
+        le_fdMonitor_Delete(gpioRef->fdMonitor);
+        gpioRef->fdMonitor = NULL;
+        int ret = 0;
+        do
+        {
+            ret = close(fd);
+        }
+        while ((ret != 0) && (errno == EINTR));
```

Comment:
I think it's invalid to clone an fd twice.
Even if an EINTR occurs, you should not be calling close on it again.

       The EINTR error is a somewhat special case.  Regarding the EINTR error, POSIX.1-2013 says:

              If close() is interrupted by a signal that is to be caught, it shall return -1 with errno set to EINTR and the state of fildes is unspecified.

       This permits the behavior that occurs on Linux and many other implementations, where, as with other errors that may be reported by close(), the file descriptor is guaranteed to be closed.   However,
       it also permits another possibility: that the implementation returns an EINTR error and keeps the file descriptor open.  (According to its documentation, HP-UX's close() does this.)  The caller must
       then once more use close() to close the file descriptor, to avoid file descriptor leaks.  This divergence in implementation behaviors provides a difficult hurdle for portable applications, since  on
       many  implementations, close() must not be called again after an EINTR error, and on at least one, close() must be called again.  There are plans to address this conundrum for the next major release
       of the POSIX.1 standard.